client/src/Utils/vendor/useDuplicateContactId.ts

```diff
@@ -1,24 +1,71 @@
+import { format } from 'date-fns';
 import { useSelector } from 'react-redux';
 
-import StoreStateType from 'redux/storeStateType';
 import logger from 'logger/logger';
+import Contact from 'models/Contact';
+import StoreStateType from 'redux/storeStateType';
 import { Service, Severity } from 'models/Logger';
+import InteractedContact from 'models/InteractedContact';
 import useCustomSwal from 'commons/CustomSwal/useCustomSwal';
 
 export interface IdToCheck {
     id?: string;
     serialId?: number 
 }
 
+type ContactId = string | undefined;
+
+const displayDateFormat = 'dd/MM/yyyy';
+
 const useDuplicateContactId = () => {
     
     const { alertWarning } = useCustomSwal();
     const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
     const userId = useSelector<StoreStateType, string>(state => state.user.id);
 
+    const findDuplicateIds = (idsToCheck: ContactId[]) => {
+        const trimmedIds: ContactId[] = idsToCheck.filter(id => id);
+        return trimmedIds.filter((id: ContactId, index: number) => trimmedIds.indexOf(id) !== index);
+    }
+
+    const checkExcelDuplicateKeys = (excelContacts: InteractedContact[], exisistingContacts: Contact[]) => {
+        const exisistingContactsIds : ContactId[]  =  exisistingContacts.map(contact => contact.idNumber).filter(id => id);
```

Comment:
this filter remove empty string from id list 
i think it can be removed because this filter also happens in findDuplicateIe Function